src/sdam/description/topology/mod.rs

```diff
@@ -329,21 +329,15 @@ impl TopologyDescription {
             self.transaction_support_status = TransactionSupportStatus::Unsupported;
         }
         if let Ok(Some(max_wire_version)) = server_description.max_wire_version() {
-            match self.topology_type {
-                TopologyType::Sharded => {
-                    // TODO RUST-122: support transactions on sharded clusters
-                    self.transaction_support_status = TransactionSupportStatus::Unsupported;
-                }
-                _ => {
-                    if max_wire_version < 7 {
-                        self.transaction_support_status = TransactionSupportStatus::Unsupported;
-                    } else {
-                        self.transaction_support_status = TransactionSupportStatus::Supported;
-                    }
-                }
+            // TODO RUST-734: Evaluate whether we should permanently support sharded transactions.
+            //                If we leave the feature as unsupported, we should revert this code.
+            if max_wire_version < 7
```

Comment:
Minor: I think this would be slightly more idiomatic with the assignment outside the if, i.e.
```
self.transaction_support_status = if ... { TransactionSupportStatus::Unsupported } else { TransactionSupportStatus::Supported };
```

Reply:
Ah yeah, that's much nicer. Don't know why we didn't have that before.